opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, (list, tuple)) and len(value) > 0:
+            return_code = self._check_sequence(value)
+            if return_code is not None:
+                logger.warning("%s in attribute value sequence", return_code)
+                return
+
         self.attributes[key] = value
 
+    @staticmethod
+    def _check_sequence(sequence: (list, tuple)) -> Optional[str]:
+        """
+        Checks if sequence items are valid and identical
+        """
+        first_element_type = type(sequence[0])
+
+        if issubclass(type(sequence[0]), Number):
```

Comment:
couldn't a second `type(sequence[0])` be avoided here by just using first_element_type?